invenio_oauthclient/handlers/utils.py

```diff
@@ -246,3 +246,43 @@ def decorated(*args, **kwargs):
         data = authorized_response()
         return f(*((data,) + args), **kwargs)
     return decorated
+
+
+def require_more_than_one_external_account(f):
+    """Require that the user has more than one external account for login.
+
+    If the user only has one linked external account and no means for logging
+    in via local credentials, the decorated function won't be executed.
+    This decorator is useful for disconnect handlers, to prevent users from
+    disconnecting their last potential means of authentication.
+    """
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        if current_user.is_anonymous:
+            return f(*args, **kwargs)
+
+        local_login_enabled = current_app.config.get(
+            "ACCOUNTS_LOCAL_LOGIN_ENABLED", True
+        )
+        password_set = current_user.password is not None
+        local_login_possible = local_login_enabled and password_set
+
+        remote_apps = current_app.config["OAUTHCLIENT_REMOTE_APPS"]
+        accounts = RemoteAccount.query.filter_by(
+            user_id=current_user.get_id()
+        ).all()
+
+        # find out all of the linked external accounts for the user
+        # that are currently configured and not hidden
+        linked_accounts = [
+            acc for acc in accounts
+            if acc.client_id in remote_apps and
+            not remote_apps[acc.client_id].get("hide", False)
```

Comment:
Question: is it possible that `emote_apps[acc.client_id]` fails?
I am thinking to this very very very uncommon scenario:
1. sysadmin configure remote app `A`
2. users login with `A`
3. after something, institutions change to remote app `B`, and remove `A` from configured remote apps
4. users login with `B` and go to the user profile, click on `Disconnect` on `A`

It might fail? But also the disconnection will fail... might be a very weird use case...

Reply:
that shouldn't be the case here, due to the filtering `if acc.client_id in remote_apps`, but i also have the feeling that this case could potentially lead to some unexpected situations generally